File: common/socket-events/chat-event.ts
export enum ChatEvent {
    SEND_MESSAGE = 'sendChatMessage',
    SEND_CHAT_HISTORY = 'sendChatHistory',
    CREATE_CHAT_ROOM = 'createChatRoom',
    GET_JOINED_ROOMS = 'getJoinedChatRooms',
    GET_OTHER_ROOMS = 'getOtherChatRooms',
    JOIN_ROOM = 'joinChatRoom',
    LEAVE_ROOM = 'leaveChatRoom',
    DELETE_ROOM = 'deleteChatRoom',
    CREATOR_UPGRADE = 'chatCreatorUpgrade',
    SEND_SOUND = 'sendSound',
    SEE_UNREAD_MESSAGES = 'seeUnreadMessages',
}

File: mobile_client/lib/constants/socket_events/chat_event.dart
enum ChatEvent {
  sendMessage("sendChatMessage"),
  sendChatHistory("sendChatHistory"),
  createChatRoom("createChatRoom"),
  getJoinedRooms("getJoinedChatRooms"),
  getOtherRooms("getOtherChatRooms"),
  joinRoom("joinChatRoom"),
  leaveRoom("leaveChatRoom"),
  deleteRoom("deleteChatRoom"),
  creatorUpgrade("chatCreatorUpgrade"),
  seeUnreadMessages("seeUnreadMessages"),
  sendSound("sendSound");

  const ChatEvent(this.name);
  final String name;
}
